023-connected-accounts/README.md

```diff
@@ -0,0 +1,303 @@
+-->
+
+At the moment, external accounts from an OAuth2 provider are connected to Appwrite via email. When a user creates an OAuth2 session, the email provided by the OAuth2 provider is looked up in Appwrite to find the the existing related user. If the user is not found, a new user is created. The problem with this is a person may have different emails in different OAuth2 providers which would result in a new user being created.
+
+Another problem with our current design is the OAuth2 information is integrated into the session. Because of that, once a session is deleted, there is no information about the OAuth2 provider that was used to create the session (and possibly user).
+
+## Design proposal (Step 2)
+
+[design-proposal]: #design-proposal
+
+<!--
+This is the technical portion of the RFC. Explain the design in sufficient detail, keeping in mind the following:
+
+- Its interaction with other parts of the system is clear
+- It is reasonably clear how the contribution would be implemented
+- Dependencies on libraries, tools, projects, or work that isn't yet complete
+- New API routes that need to be created or modifications to the existing routes (if needed)
+- Any breaking changes and ways in which we can ensure backward compatibility.
+- Use Cases
+- Goals
+- Deliverables
+- Changes to documentation
+- Ways to scale the solution
+
+Ensure that you include examples and code snippets to allow the community to understand the proposed solution. **It would be best if the examples use naming conventions that you intend to use during the actual implementation to suggest changes early on during the development.**
+
+Write your answer below.
+
+-->
+
+A user should be able to:
+
+1. Connect their account to an OAuth2 provider
+2. List their connected accounts
+3. Update their connected account
+   * This will refresh the access token
+4. Disconnect their account from an OAuth2 provider
+5. Create an OAuth2 session with an existing account without a connected OAuth2 provider
+   * This will automatically create a connected account
+6. Create an OAuth2 session without creating an account (backwards compatibility)
+
+A developer should be able to:
+
+1. List a user's connected accounts
+2. Delete a user's connected account
+
+### API Endpoints
+
+<!--
+List the new API routes or endpoints that we might need to add for supporting the new feature.
+Keep in mind to stay very strict to the API protocol and method, whether your new
+changes are for the REST, WebSocket or any other API protocol Appwrite supports.
+
+For example:
+
+**POST /v1/coffee ** - an endpoint for creating coffee.
+**DELETE /v1/coffee ** - an endpoint for deleting coffee.
+-->
+
+#### Account Service
+
+##### Create Connected Account
+
+The existing Create OAuth2 Session endpoints will be re-used to create a connected account
+
+**GET /v1/account/sessions/oauth2/{provider}**
+
+##### List Connected Accounts
+
+**GET /v1/account/connected**
```

Comment:
```suggestion
**GET /v1/account/oauth2/connections**
```

Connected accounts is very misleading. When I first heard of it I thought it was about connecting 2 existing Appwrite accounts and allowing to switch between them. This features is actually just refactoring of how we handle OAuth2.  Also, the `connected` is a past tense verb where a REST API should be represented by nouns.

Reply:
Wouldn't it be useful to make this flexible so that we could support other connections? Maybe multiple phone numbers or linking multiple emails? Merging Appwrite users can also be helpful.